dialogue-clients/src/main/java/com/palantir/dialogue/clients/DialogueDnsResolutionWorker.java

```diff
@@ -64,15 +70,23 @@ public void run() {
         while (!shutdownRequested) {
             try {
                 // check for updates to scb state first
-                Thread.sleep(5000L);
+                Thread.sleep(dnsRefreshInterval.toMillis());
                 try {
+                    if (receiver.get() == null) {
+                        shutdownRequested = true;
+                        log.info("Output refreshable has been garbage collected, no need to continue polling");
```

Comment:
should there be an `else` or a `break` here?

Reply:
Ya, we should probably add an `else` here to avoid unnecessary work. `doUpdate` must always handle the not-present case either way, as the gc may run between doing a pre-check and later on.